CRM/Case/BAO/Case.php

```diff
@@ -529,8 +529,11 @@ public static function retrieveCaseIdsByContactId($contactID, $includeDeleted =
     if (!$includeDeleted) {
       $query .= " AND ca.is_deleted = 0";
     }
+    if (isset($statusId)) {
+      $query .= " AND ca.status_id = %2";
```

Comment:
That looks fine, although there is no good reason (and it may cause problems) to include $statusId in the $params array if it is not set, so I would move the $params definition to above this line, and append to the array inside this conditional.